plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRestCatalog.java

```diff
@@ -211,4 +212,31 @@ public void testPrefix()
                 .as("should fail as the prefix dev is not implemented for the current endpoint")
                 .hasMessageContaining("Malformed request: No route for request: POST v1/dev/namespaces");
     }
+
+    @Test
+    void testNestedNamespace()
+    {
+        TrinoCatalog catalog = createTrinoCatalog(false);
+
+        String firstLevel = "level-1" + randomNameSuffix();
+        String secondLevel = "level-2" + randomNameSuffix();
+        String thirdLevel = "level-3" + randomNameSuffix();
+
+        List<String> namespaces = ImmutableList.<String>builder()
+                .add(firstLevel)
+                .add(firstLevel + "." + secondLevel)
+                .add(firstLevel + "." + secondLevel + "." + thirdLevel)
+                .build();
+
+        namespaces.forEach(namespace -> catalog.createNamespace(SESSION, namespace, ImmutableMap.of(), new TrinoPrincipal(PrincipalType.USER, SESSION.getUser())));
+
+        try {
+            assertThat(catalog.listNamespaces(SESSION))
+                    .hasSize(namespaces.size())
+                    .containsAll(namespaces);
+        }
+        finally {
+            namespaces.forEach(namespace -> catalog.dropNamespace(SESSION, namespace));
```

Comment:
`DelegatingRestSessionCatalog` allows dropping parent namespace without it being empty. If we use `TestingPolarisCatalog` to create `RESTSessionCatalog` then, drop ns fails as dropping parent ns is not allowed when its non-empty. It may not be worth using TestingPolarisCatalog here but we can reverse list while dropping like `namespaces.reversed().forEach(namespace -> catalog.dropNamespace(SESSION, namespace));` to show intent that child namespaces are dropped before parent ns.

Reply:
It implies this test isn't helpful at this time. I would recommend removing instead.

Reply:
Sounds good to me. Additionally,  `TestIcebergRestCatalogNestedNamespaceConnectorSmokeTest#testDropNestedSchemaCascade` tests similar logic.